src/app/seamlessupdate/client/PeriodicJob.java

```diff
@@ -25,16 +29,20 @@ static void schedule(final Context context) {
                 jobInfo.getNetworkType() == networkType &&
                 jobInfo.isRequireBatteryNotLow() == batteryNotLow &&
                 jobInfo.isPersisted() &&
-                jobInfo.getIntervalMillis() == INTERVAL_MILLIS) {
+                jobInfo.getIntervalMillis() == INTERVAL_MILLIS &&
+                jobInfo.getExtras().getString(EXTRA_JOB_CHANNEL).equals(channel)) {
```

Comment:
Isn't it possible for `getString` to be `null`, such as right after an update introducing this change?